packages/react/src/providers/helio/HelioContext.tsx

```diff
@@ -79,27 +90,38 @@ export const useHelioProvider = () => {
     }
   };
 
+  const getPaymentDetails = <T extends PaymentDetailsType>(): T =>
+    paymentDetails as T;
+
+  const getPaymentFeatures = <T extends PaymentFeatures>(): T =>
+    getPaymentDetails()?.features as T;
+
   const checkCustomerDetailsRequired = (): boolean => {
     if (!paymentDetails) return false;
+
     return !!(
-      paymentDetails.features?.requireEmail ||
-      paymentDetails.features?.requireFullName ||
-      paymentDetails.features?.requireDiscordUsername ||
-      paymentDetails.features?.requireTwitterUsername ||
-      paymentDetails.features?.requireCountry ||
-      paymentDetails.features?.requireDeliveryAddress ||
-      paymentDetails.features?.canChangeQuantity || // @TODO remove
-      paymentDetails.features?.canChangePrice
+      getPaymentFeatures()?.requireEmail ||
```

Comment:
why is this a function ? can't you just return an object ? even so then maybe it's better to call it once